os/include/tinyara/lcd/st7785.h

```diff
@@ -59,28 +57,81 @@
+        {0xB2, 10, {0x00, 0x0C, 0x00, 0x0C, 0x00, 0x00, 0x00, 0x33, 0x00, 0x33}}, /* Porch Setting */ \
+        {0xB7, 2, {0x00, 0x51}},	/* Gate Control */                                            \
+        {0xBB, 2, {0x00, 0x22}},	/* VCOMS Setting */                                           \
+        {0xC0, 2, {0x00, 0x2C}},	/* LCM Control */                                             \
+        {0xC2, 2, {0x00, 0x01}},	/* VRH Command Enable */                                      \
+        {0xC3, 2, {0x00, 0x13}},	/* VRH Set */                                                 \
+        {0xC6, 2, {0x00, 0x0F}},	/* Frame Rate Control in Normal Mode */                       \
+        {0xD0, 2, {0x00, 0xA7}},	/* Power Control 1 */                                         \
+        {0xD0, 4, {0x00, 0xA4, 0x00, 0xA1}}, /* Power Control 1 */                                    \
+        {0xE0, 28, {0x00, 0xF0, 0x00, 0x00, 0x00, 0x04, 0x00, 0x03, 0x00, 0x05, 0x00, 0x04, 0x00, 0x2E, 0x00, 0x44, 0x00, 0x45, 0x00, 0x39, 0x00, 0x14, 0x00, 0x14, 0x00, 0x2D, 0x00, 0x35}}, /* Positive Voltage Gamma Control */ \
+        {0xE1, 28, {0x00, 0xF0, 0x00, 0x0A, 0x00, 0x0E, 0x00, 0x0F, 0x00, 0x0B, 0x00, 0x26, 0x00, 0x2E, 0x00, 0x43, 0x00, 0x45, 0x00, 0x36, 0x00, 0x12, 0x00, 0x12, 0x00, 0x2A, 0x00, 0x32}}, /* Negative Voltage Gamma Control */ \
+        {0x2A, 8, {0x00, 0x00, 0x00, 0x00, 0x00, LCD_CASET1, 0x00, LCD_CASET2}},	/* Column Address Set */ \
+        {0x2B, 8, {0x00, 0x00, 0x00, 0x00, 0x00, LCD_RASET1, 0x00, LCD_RASET2}},	/* Row Address Set */    \
+        {0x21, 0, {0x00}},	/* Display Inversion On */
+
+static const lcm_setting_table_t lcd_init_cmd_g_avd[] = {
         {0x11, 0, {0x00}},		/* Sleep out */
         {REGFLAG_DELAY, 120, {}},	/* Delayms (120) */
         {0x36, 2, {0x00, LCD_ORIENTATION}},	/* Memory Data Access Control */
-        {0x3A, 2, {0x00, 0x55}},	/* Interface Pixel format 0x55 for RGB565 */
-        {0xB0, 2, {0x00, 0x10}},	/* RAM control */
+        ST7785_COMMON_INIT
+        {0x29, 0, {0x00}},	/* Display On */
+        {REGFLAG_END_OF_TABLE, 0, {}},
+};
+
+static const lcm_setting_table_t lcd_init_cmd_g_holitech[] = {
+        {0x36, 2, {0x00, LCD_ORIENTATION}},	/* Memory Data Access Control */
+        ST7785_COMMON_INIT
+        {0x11, 0, {0x00}},		/* Sleep out */
+        {REGFLAG_DELAY, 120, {}},	/* Delayms (120) */
+        {0x29, 0, {0x00}},	/* Display On */
+        {REGFLAG_DELAY, 10, {}},	/* Delayms (10) */
+        {REGFLAG_END_OF_TABLE, 0, {}},
+};
+
+// Vendor mapping table
+typedef struct {
+        uint8_t id[LCD_ID_LENGTH];
+        lcm_setting_table_t *init_cmd;
+} lcd_vendor_map_t;
+
+lcd_vendor_map_t g_lcd_vendors[] = {
+        {{0x8A, 0x8A, LCD_AVD}, lcd_init_cmd_g_avd},      // AVD
+        {{LCD_HOLITECH, 0x8A, 0x85}, lcd_init_cmd_g_holitech}, // Holitech
+};
+#define NUM_LCD_VENDORS (sizeof(g_lcd_vendors) / sizeof(g_lcd_vendors[0]))
+#else
```

Comment:
How about intergrate A and B like below?
It will make to support several lcd type
```c
typedef struct {
        uint8_t id[LCD_ID_LENGTH];
        lcm_setting_table_t *init_cmd;
} lcd_vendor_map_t;

lcd_vendor_map_t g_lcd_vendors[] = {
#ifdef CONFIG_LCD_ST7785_TYPEA
        {{0x8A, 0x8A, LCD_AVD}, lcd_init_cmd_g},      // AVD A
#endif
#ifdef CONFIG_LCD_ST7785_TYPEB
        {{0x8A, 0x8A, LCD_AVD}, lcd_init_cmd_g_avd},      // AVD B
        {{LCD_HOLITECH, 0x8A, 0x85}, lcd_init_cmd_g_holitech}, // Holitech
#endif
};
#define NUM_LCD_VENDORS (sizeof(g_lcd_vendors) / sizeof(g_lcd_vendors[0]))


```